error/s2n_errno.c

```diff
@@ -263,6 +263,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_INVALID_X509_EXTENSION_TYPE, "Invalid X509 extension type") \
     ERR_ENTRY(S2N_ERR_INSUFFICIENT_MEM_SIZE, "The provided buffer size is not large enough to contain the output data. Try increasing the allocation size.") \
     ERR_ENTRY(S2N_ERR_KEYING_MATERIAL_EXPIRED, "The lifetime of the connection keying material has exceeded the limit. Perform a new full handshake.") \
+    ERR_ENTRY(S2N_ERR_REJECTED_EARLY_DATA, "Unable to decrypt rejected early data") \
```

Comment:
Is the only scenario for using this error code when we fail to decrypt the early data? If so, we could be more precise in the error code. If we don't want to be precise, consider removing the `decrypt` from the description and say `Rejected early data`?